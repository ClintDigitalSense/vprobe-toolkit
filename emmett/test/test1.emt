HV_Exit tsc = TSC;
HV_Resume ag[VMCS_CS & 3, VMCS_RIP] <- TSC - tsc;
VMM1Hz {
   /* Blocks work! */
   bits = 12;
   operatorPrecedence = 1 + 2 * 3 > 12 && ~bits == 3;
   if (operatorPrecedence == 3 && bits==0xdeadbeef) {
      foo=bits;
   }
}
