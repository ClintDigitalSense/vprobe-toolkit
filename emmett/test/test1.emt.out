(version 0.2)
(defaggr ag 1 0)
(definteger bits)
(definteger foo)
(definteger operatorPrecedence)
(definteger tsc)

 ;;;;;
(vprobe HV_Exit ( setint tsc   TSC))
 ;;;;;
(vprobe HV_Resume ( aggr  ag (  (& VMCS_CS  3 )) ()  (- TSC  tsc )))
 ;;;;;
(vprobe VMM1Hz ( do  ( setint bits   12)( setint operatorPrecedence    (&&  (> 7  12 )   (==  (~ bits )  3 ) ))( cond  ( (&&  (== operatorPrecedence  3 )   (== bits  3735928559 ) )( do  ( setint foo   bits))))))